functorch/test/test_dims.py

```diff
@@ -0,0 +1,592 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from functorch.dim import Tensor, Dim, dims, dimlists, stack, DimensionBindError, DimList
+
+from attn_ft import BertSelfAttention as BertSelfAttentionA, Linear
+from attn_positional import BertSelfAttention as BertSelfAttentionB
+
+from unittest import TestCase, main, skip
+import torch
+import gc
+
+from functorch._C import dim as _C
+
+try:
+    from torchvision.models import resnet18
+except ImportError:
+    resnet18 = None
+
+_test_c, _parse_test, _set_pointwise_optimize = _C._test_c, _C._parse_test, _C._set_pointwise_optimize
+
+from contextlib import contextmanager
+from time import perf_counter
+
+measure_perf = False
+if measure_perf:
+    from torchdim.magic_trace import magic_trace
+else:
+    @contextmanager
+    def magic_trace(*args, **kwargs):
+        yield
+
+@contextmanager
+def measure(what):
+    b = perf_counter()
+    yield
+    e = perf_counter()
+    print(f"{what}: {e - b:.20f} seconds")
+
+def triu(A):
+    i, j = dims()
+    a = A[i, j]
+    zero = torch.tensor(0, dtype=torch.float)  # XXX - torch.where is janky...
+    return torch.where(i <= j, a, zero).order(i, j)
+
+def gpu_time(lmb, name, r=100):
+    b = torch.cuda.Event(enable_timing=True)
+    e = torch.cuda.Event(enable_timing=True)
+    # with magic_trace(name + ".fxt"):
+    for _ in range(r):
+        lmb()
+    b.record()
+    for _ in range(r):
+        lmb()
+    e.record()
+    e.synchronize()
+    elapsed = b.elapsed_time(e)
+    # with torch.profiler.profile(schedule=torch.profiler.schedule(
+    #     wait=0,
+    #     warmup=1,
+    #     active=2), on_trace_ready=tensorboard_trace_handler(name), with_stack=True) as profiler:
+    #     for _ in range(3):
+    #         lmb()
+    #         profiler.step()
+    print(name, elapsed / r)
+    return elapsed / r
+
+class TestMin(TestCase):
```

Comment:
For a test to be runnable via the pytorch testing infrastructure (i.e. run_test.py), it needs to extend from torch.testing._internal.common_utils.TestCase. That might just be a drop-in replacement for unittest.TestCase here